sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs

```diff
@@ -935,126 +920,62 @@ private async Task RunAsync(CancellationToken cancellationToken)
             {
                 var cycleDuration = Stopwatch.StartNew();
 
-                // Renew this instance's ownership so they don't expire.
-
-                await RenewOwnershipAsync(cancellationToken).ConfigureAwait(false);
-
                 // Some previously owned partitions might have had their ownership expired or might have been stolen, so we need to stop
                 // the processing tasks we don't need anymore.
 
                 await Task.WhenAll(ActivePartitionProcessors.Keys
-                    .Except(InstanceOwnership.Keys)
+                    .Except(LoadBalancer.OwnedPartitionIds)
                     .Select(partitionId => StopPartitionProcessingIfRunningAsync(partitionId, ProcessingStoppedReason.OwnershipLost, cancellationToken)))
                     .ConfigureAwait(false);
```

Comment:
Good question. I wondered about the ordering here also and forgot to come back to it. 

I'm thinking it would be better to keep `RenewOwnership` encapsulated inside the load balancer.  How do you feel about moving this processing task cleanup logic to the end of the cycle instead?

Reply:
The only problem I see with this is renewing an onwership that the processor has just claimed. The processor will be balanced most of the time, so this will happen so few times that I think it's not worth bothering about. I'm totally okay with it.

@jsquire I'm not sure how concerned we are about service calls. Do you think it's worrisome to keep it the way it is?

Reply:
I'm not concerned with a few extra network requests that would be needed for this; we'll need to be somewhat chatty by nature to coordinate the load balancing anyway, so we're talking just a small and occasional extra request or three on top of that.   Please let me know if I'm understating that, but I think my context is correct.

Reply:
You got it right. We should keep it the way it is, then!